### DISCUSSION TITLE
Content Delivery path to be constant

### DISCUSSION THREAD

**Discussion by andr-04** [tolgee/tolgee-platform#2287] [Q&A] [answered]
Hello.

I started Tolgee on a self-hosted mode few days ago and faced the problem.

Each time I save settings of [Content Delivery](https://tolgee.io/platform/projects_and_organizations/content_delivery) to the external S3 CDN, the path is changed. It is very unuseful because it requires to update the pull address on the software to download localization results from.

How to lock the path address of Content Delivery?

**Comment by JanCizmar** [comment]
Hey! Thanks for reporting. How exactly are you changing the content delivery settings? Is it via the UI or API?

**Reply by andr-04** [reply]
Via the UI.

For example, I plan adding new languages. Now it requires updating references to all other languages too, and removing old (outdated) directories from S3 by hand.

**Reply by JanCizmar** [reply]
I just tried to reproduce it, but it looks OK. Can you please attach a video showing the reported behaviour?

**Reply by andr-04** [reply]
https://github.com/tolgee/tolgee-platform/assets/2203770/1bd0c458-cfd6-4a89-a42e-d2e1752dfd94

**Reply by JanCizmar** [reply]
Oh. I see, you are using server-level storage. My bad. It's a bug. Thanks for reporting.

**Reply by JanCizmar** [reply]
This is fixing PR: https://github.com/tolgee/tolgee-platform/pull/2288

**Reply by JanCizmar** [reply, marked as answer]
It's fixed in v3.59.5
